src/MonoTorrent/MonoTorrent.Client.Connections/SocketConnection.cs

```diff
@@ -198,16 +198,20 @@ public ReusableTask<int> ReceiveAsync (byte[] buffer, int offset, int count)
             args.SetBuffer (offset, count);
             args.UserToken = ReceiveTcs;
 
+            // https://github.com/alanmcgovern/monotorrent/issues/202
+#if DEBUG
             AsyncFlowControl? control = null;
             if (!ExecutionContext.IsFlowSuppressed ())
                 control = ExecutionContext.SuppressFlow ();
-
+#endif
```

Comment:
If this is wrapped in `#if DEBUG` then it will only have an effect when running local debug builds. The NuGet packages from nuget.org are built with the `Release` configuration so this code will still execute. Is this intended?

Also, this changeset would be good as a separate PR.

Reply:
A similar change has been integrated and released in the latest nuget package. There is a bug in some older versions of UWP which causes nullreference exceptions if you use this API. I assume you encountered that issue :)